models/Asset/Listing/Dao.php

```diff
@@ -99,10 +99,11 @@ public function loadIdList()
      */
     public function getCount()
     {
-        $select = $this->getQuery([new Expression('COUNT(*)')]);
-        $amount = (int) $this->db->fetchOne($select, $this->model->getConditionVariables(), $this->model->getConditionVariableTypes());
+        if (count($this->model->getAssets()) == 0) {
```

Comment:
isn't that way slower? Cause it needs to hydrate the models.

Reply:
same approach as for objects. at least it produces the correct result

Reply:
Why don't you use `getQuery` and change it up to make a COUNT(*) query?

https://github.com/pimcore/pimcore/blob/9167b96ed44aef0d5f029be73d120faec972f3a8/models/Asset/Listing/Dao.php#L64

Reply:
just realised, that is exactly what has been proposed in #5639